olive/workflows/run/run.py

```diff
@@ -44,7 +47,61 @@ def automatically_insert_passes(config):
     return new_engine, new_config
 
 
-def run(config: Union[str, Path, dict]):
+def dependency_setup(config):
+    here = os.path.abspath(os.path.dirname(__file__))
+    EXTRAS = json.load(open(os.path.join(here, "../../extra_dependencies.json"), "r"))
+    DEPENDENCY_MAPPING = {
+        "device": {
+            SystemType.AzureML: EXTRAS.get("azureml"),
+            SystemType.Docker: EXTRAS.get("docker"),
+            SystemType.Local: {Device.CPU: EXTRAS.get("cpu"), Device.GPU: EXTRAS.get("gpu")},
+        },
+        "pass": {
+            "OnnxFloatToFloat16": ["onnxconverter-common"],
+            "OrtPerfTuning": ["psutil"],
+            "QuantizationAwareTraining": ["pytorch-lightning"],
+            "OpenVINOConversion": EXTRAS.get("openvino"),
+            "OpenVINOQuantization": EXTRAS.get("openvino"),
+        },
+    }
+
+    required_packages = []
+
+    for _, pass_config in config.passes.items():
+        if DEPENDENCY_MAPPING["pass"].get(pass_config.type):
+            required_packages.extend(DEPENDENCY_MAPPING["pass"].get(pass_config.type))
+        if pass_config.type in ["SNPEConversion", "SNPEQuantization", "SNPEtoONNXConversion"]:
+            logger.info(
+                "Please refer to https://microsoft.github.io/Olive/tutorials/passes/snpe.html \
+                to install SNPE Prerequisites for pass {}".format(
+                    pass_config.type
+                )
+            )
+
+    if config.engine.host.type == SystemType.Local:
+        required_packages.extend(
+            DEPENDENCY_MAPPING["device"][config.engine.host.type][config.engine.host.config.device]
+        )
+        logger.info("The following packages will be installed: {}".format(" ".join(required_packages)))
+        for package in set(required_packages):
+            try:
+                __import__(package)
+            except (ImportError):
+                subprocess.check_call(["pip", "install", "{}".format(package)])
+    else:
+        for package in set(DEPENDENCY_MAPPING["device"][config.engine.host.type]):
+            try:
+                __import__(package)
+            except (ImportError):
+                subprocess.check_call(["pip", "install", "{}".format(package)])
+        logger.info(
```

Comment:
Looks like the info is telling that they should install the packages. But it already finished the installation in the position that you are logging.

Reply:
the packages installed are packages needed for pipelines, like azureml or docker package. The packages in log are required for passes, for example if user wants to run perf tuning on aml, he needs to have psutil in environment config file.